File: clients/ruby/model/loot_box.rb
require_relative 'vec2_double'
require_relative 'vec2_double'
require_relative 'item'
class LootBox
    attr_accessor :position
    attr_accessor :size
    attr_accessor :item
    def initialize(position, size, item)
        @position = position
        @size = size
        @item = item
    end
    def self.read_from(stream)
        position = Vec2Double.read_from(stream)
        size = Vec2Double.read_from(stream)
        item = Item.read_from(stream)
        LootBox.new(position, size, item)
    end
    def write_to(stream)
        @position.write_to(stream)
        @size.write_to(stream)
        @item.write_to(stream)
    end
end

File: clients/go/model/loot_box.go
package model

import "io"

type LootBox struct {
	Position *Vec2Float64
	Size     *Vec2Float64
	Item     IItem
}

func NewLootBox(position *Vec2Float64, size *Vec2Float64, item IItem) *LootBox {
	return &LootBox{
		Position: position,
		Size:     size,
		Item:     item,
	}
}
func ReadLootBox(reader io.Reader) *LootBox {
	return &LootBox{
		Position: ReadVec2Float64(reader),
		Size:     ReadVec2Float64(reader),
		Item:     ReadItem(reader),
	}
}
func (value *LootBox) Write(writer io.Writer) {
	value.Position.Write(writer)
	value.Size.Write(writer)
	value.Item.Write(writer)
}
